Source/Dafny/Compiler-java.cs

```diff
@@ -1978,7 +1978,9 @@ private class GenericArrayElementLvalue : ILvalue {
 
     protected void EmitToString(TargetWriter wr, Expression arg) {
```

Comment:
Judging from the uses of `EmitToString` in Compiler-java.cs, I'm thinking the fix in this PR should perhaps also be applied `expect` statements. Or, please add tests for `expect` as well, to make sure there are no cast problems there.

Reply:
Expect takes a boolean -- so are you thinking of general casts in expressions? I'll make a separate Issue for expect (and assert and assume?) if so.